### DISCUSSION TITLE
How to enable ClusterVulnerabilityReports?

### DISCUSSION THREAD

**Discussion by sudoleg** [aquasecurity/trivy-operator#1898] [Help & Support] [answered]
Hello folks, I've deployed and configured the operator via Helm. It successfully generates Vulnerability-, ConfigAudit and InfraAssessment Reports, but I am missing the [ClusterVulnerabilityReports](https://aquasecurity.github.io/trivy-operator/latest/docs/crds/clustervulnerability-report/). The ClusterVulnerability CRD is installed, but there are no resources based on it. I also can't find a corresponding 'enabling' option in the values... How can I enable this kind of report when deploying the operator via Helm?

Thanks in advance :)

**Comment by chen-keinan** [comment]
@sudoleg are you using native k8s or cloud ? currently only native or RKe2 clusters are supported

**Reply by sudoleg** [reply]
I am using a native k8s cluster

**Reply by chen-keinan** [reply]
which installation (vendor) are you using ?

**Reply by sudoleg** [reply]
we are using a completely self-provisioned cluster, basically "vanilla Kubernetes"

**Comment by chen-keinan** [comment]
@sudoleg do you mind sharing your configurations  ?

**Reply by sudoleg** [reply]
No problem. FYI, we deploy the operator and the standalone trivy (server) in the namespace called `trivy-system`. I attached the contents of the configmaps for the operator and the server. I suppose all the values that aren't displayed in the configmaps are defaullts.

[trivy-operator-trivy-config.json](https://github.com/aquasecurity/trivy-operator/files/14556399/trivy-operator-trivy-config.json)
[trivy-operator-trivy-server.json](https://github.com/aquasecurity/trivy-operator/files/14556403/trivy-operator-trivy-server.json)

Hope those configs are the ones you requested

**Reply by chen-keinan** [reply]
@sudoleg I'm looking specific to see if the following two envvar are set:
```
OPERATOR_SBOM_GENERATION_ENABLED: true
OPERATOR_VULNERABILITY_SCANNER_ENABLED: true
```

**Reply by sudoleg** [reply]
The SBOM Generation is disabled. Do wen need this to enable ClusterVulnerability Reports?

**Reply by chen-keinan** [reply, marked as answer]
yes it is required.

**Reply by sudoleg** [reply]
I've enabled SBOM generation and now there is a ClusterVulnerability report, but it's empty for some reason (the `vulnerabilities` list is empty). I can't figure out why... Do you have an idea? @chen-keinan

**Reply by chen-keinan** [reply]
maybe your cluster do not have vulnerabilities , its only check vulnerabilites on core components (api-server , kubelet, controller-manager and etc)
see [vulndb](https://github.com/aquasecurity/vuln-list-k8s/tree/main/upstream) for k8s here

**Reply by sudoleg** [reply]
Hmmm ok... didn't expect that a cluster with 7 nodes based on Ubuntu 20.04 has no vulnerabilities (at least not listed in the repo you provided) but it seems like that 🤷‍♂️

Thank you for the quick respose! 👍 

I think, it's not quite obvious from the docs/description that you have to enable SBOMs to activate ClusterVulnerabilityReports... May I submit a PR to adjust this Part of the documentation?
